cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -381,10 +381,18 @@ private void createHeartbeatsFeed() throws NotificationFeedException {
       .setDescription("Stream heartbeats feed.")
       .build();
 
-    try {
-      feedManager.getFeed(streamHeartbeatsFeed);
-    } catch (NotificationFeedNotFoundException e) {
-      feedManager.createFeed(streamHeartbeatsFeed);
+    while (true) {
+      try {
+        feedManager.getFeed(streamHeartbeatsFeed);
+        return;
+      } catch (NotificationFeedNotFoundException e) {
+        feedManager.createFeed(streamHeartbeatsFeed);
+        return;
+      } catch (NotificationFeedException e) {
+        // Most probably, the dataset service is not up. We retry
+        LOG.warn("Could not subscribe to heartbeats feed. {}", e.getMessage());
```

Comment:
think you can just give it the exception ```LOG.warn(" .... ", e)```. The message is also incorrect, should be something about being unable to create the feed.